Dockerfile

```diff
@@ -1,23 +1,51 @@
-FROM node:12-buster
-
-RUN set -e; \
-    apt update; \
-    apt install -y gettext; \
-    rm -rf /var/lib/apt/lists/*
+FROM node:12-alpine AS BUILD_IMAGE
 
 ARG branch=master
 
 ENV NODE_ENV production
 WORKDIR /opt/magic_mirror
 
-RUN git clone --depth 1 -b ${branch} https://github.com/MichMich/MagicMirror.git .
-RUN cp -R modules /opt/default_modules
-RUN cp -R config /opt/default_config
-RUN npm install --unsafe-perm --silent
+# get magic mirror
+RUN apk update && apk add --no-cache git \
+    && git clone --depth 1 -c advice.detachedHead=false -b ${branch} https://github.com/MichMich/MagicMirror.git . \
+    && apk del git \
+    && rm -rf /var/cache/apk/*
+
+# save default modules and configuration and install dependencies
+RUN cp -R modules /opt/magic_mirror/default_modules \
+    && cp -R config /opt/magic_mirror/default_config \
+    && npm set unsafe-perm true \
+    && npm ci \
+    # && npm prune --production
+    && wget -q https://install.goreleaser.com/github.com/tj/node-prune.sh | sh
+
+
+
+FROM node:12-alpine
+
+WORKDIR /opt/magic_mirror
+
+RUN set -x \
+    && apk add --update libintl \
+    && apk add --virtual build_deps gettext \
+    && apk del build_deps
+
+COPY --from=BUILD_IMAGE /opt/magic_mirror/node_modules ./node_modules
+COPY --from=BUILD_IMAGE /opt/magic_mirror/config ./config
+COPY --from=BUILD_IMAGE /opt/magic_mirror/css ./css
+COPY --from=BUILD_IMAGE /opt/magic_mirror/fonts ./fonts
+COPY --from=BUILD_IMAGE /opt/magic_mirror/js/ ./js
+COPY --from=BUILD_IMAGE /opt/magic_mirror/serveronly ./serveronly
+COPY --from=BUILD_IMAGE /opt/magic_mirror/translations ./translations
+COPY --from=BUILD_IMAGE /opt/magic_mirror/vendor ./vendor
+COPY --from=BUILD_IMAGE /opt/magic_mirror/index.html /opt/magic_mirror/package.json /opt/magic_mirror/package-lock.json ./
+COPY --from=BUILD_IMAGE /opt/magic_mirror/default_config ./default_config
+COPY --from=BUILD_IMAGE /opt/magic_mirror/default_modules/ ./default_modules
```

Comment:
instead of this big block here, we could prepared a directory in the builder image and only do one copy operation.

Reply:
Happy if you have advice on how to do that. I will also look up if the copying can be optimized

Reply:
```suggestion
COPY --from=BUILD_IMAGE /dist/ .
```